packages/react-instantsearch-hooks-dom/src/ui/ToggleRefinement.tsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+
+import { cx } from './lib/cx';
+
+export type ToggleRefinementClassNames = {
+  /**
+   * Class names to apply to the root element
+   */
+  root: string;
+  /**
+   * Class names to apply to the label element
+   */
+  label: string;
+  /**
+   * Class names to apply to the checkbox element
+   */
+  checkbox: string;
+  /**
+   * Class names to apply to the label text
+   */
+  labelText: string;
+};
+
+export type ToggleRefinementProps = Omit<
+  React.HTMLAttributes<HTMLDivElement>,
+  'onChange'
+> &
+  Pick<React.InputHTMLAttributes<HTMLInputElement>, 'checked'> & {
+    classNames?: Partial<ToggleRefinementClassNames>;
+    onChange(isChecked: boolean): void;
+    label: string;
+  };
+
+export function ToggleRefinement({
+  classNames = {},
+  checked,
+  onChange,
+  label,
+  ...props
+}: ToggleRefinementProps) {
+  return (
+    <div
+      {...props}
+      className={cx('ais-ToggleRefinement', classNames.root, props.className)}
+    >
+      <label className={cx('ais-ToggleRefinement-label', classNames.label)}>
+        <input
+          className={cx('ais-ToggleRefinement-checkbox', classNames.checkbox)}
+          type="checkbox"
+          checked={checked}
+          onChange={(event) => {
+            onChange(event.target.checked);
```

Comment:
in InstantSearch.js this is:

```suggestion
            onChange({ checked: event.target.checked });
```

I guess typescript explaining what the boolean can also help, but a loose boolean can be confusing

Reply:
Yes, I kept this part in the widget so we're not leaking the API of `useToggleRefinement`. I think this is fine since this is a UI component, users don't interact with it.